chat_app_be/chat_app_be/Services/UserService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using chat_app_be.Models.Auth;
+using chat_app_be.Models;
+using chat_app_be.Repositories.Interfaces;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using chat_app_be.Dtos;
+using chat_app_be.Models.Response;
+using AutoMapper;
+using chat_app_be.Services.Interfaces;
+using Microsoft.Extensions.Options;
+
+namespace chat_app_be.Services
+{
+    public class UserService : IUserService
+    {
+        private readonly IUserRepository _userRepository;
+        private readonly IOptions<JwtOptions> _jwtOptions;
+        private readonly IMapper _mapper;
+        public UserService(IUserRepository userRepository, IOptions<JwtOptions> jwtOptions, IMapper mapper)
+        {
+            _userRepository = userRepository;
+            _jwtOptions = jwtOptions;
+            _mapper = mapper;
+        }
+
+        public async Task<Response> Register(UserDto user)
+        {
+            try
+            {
+                if (await _userRepository.UserExistsAsync(user.Username))
+                {
+                    return new Response(statusCodes: StatusCodes.Status400BadRequest, "Username Already Exists");
+                }
+
+                CreatePasswordHash(user.Password, out byte[] passwordHash, out byte[] passwordSalt);
+
+                var newUser = new User
+                {
+                    Username = user.Username,
+                    DisplayName = user.DisplayName,
+                    PasswordHash = passwordHash,
+                    PasswordSalt = passwordSalt
+                };
+
+                await _userRepository.AddUserAsync(newUser);
+
+                UserResponseDto userResponse = _mapper.Map<UserResponseDto>(newUser);
+
+                return new Response(statusCodes: StatusCodes.Status200OK, userResponse);
+            }
+            catch (Exception e)
+            {
+                return new Response(statusCodes: StatusCodes.Status500InternalServerError, "Something is Wrong");
+            }
+        }
+
+        public async Task<Response> Login(UserRequestDto request)
+        {
+            try
+            {
+                var user = await _userRepository.GetUserByUsernameAsync(request.Username);
+
+                if (user == null)
+                {
+                    return new Response(statusCodes: StatusCodes.Status404NotFound, "User Not Found"); ;
+                }
+
+                if (!VerifyPasswordHash(request.Password, user.PasswordHash, user.PasswordSalt))
+                {
+                    return new Response(statusCodes: StatusCodes.Status400BadRequest, "Wrong Password"); ;
+                }
+
+                string token = CreateToken(user);
+
+                return new Response(statusCodes: StatusCodes.Status200OK, data: token);
```

Comment:
Instead of returning only the token, we could return additional user information such as first name, last name, etc., which would be useful for further actions.